src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java

```diff
@@ -142,4 +198,64 @@ public SQLConnection getTransactionIsolation(Handler<AsyncResult<TransactionIsol
     conn.schedule(new PreparedTxQueryCommand(handler));
     return this;
   }
+
+  @Override
+  public void close(Handler<AsyncResult<Void>> handler) {
+    inAutoCommit.set(true);
+    if (inTransaction.compareAndSet(true, false)) {
+      conn.schedule(new QueryCommand("COMMIT", new ResultSetBuilder(re -> {
+        if(re.failed()) {
+          handler.handle(Future.failedFuture(re.cause()));
+        } else {
+          // should we use close command for open portal if any ?
+          conn.doClose();
+          handler.handle(Future.succeededFuture());
+        }
+      })));
+    } else {
+      conn.doClose();
+      handler.handle(Future.succeededFuture());
+    }
+  }
+
+  @Override
+  public void close() {
+    close(c -> {
+    });
+  }
+
+  private void beginTxIfNeeded(Handler<AsyncResult<Void>> handler) {
+    if (!inAutoCommit.get() && inTransaction.compareAndSet(false, true)) {
+      conn.schedule(new QueryCommand("BEGIN", new ResultSetBuilder(re -> {
```

Comment:
This could be enhanced to support TX isolation levels, maybe a follow up pull request see: https://github.com/vert-x3/vertx-mysql-postgresql-client/pull/74/files#diff-7a3a21a56ba64de74ca084f12538d869